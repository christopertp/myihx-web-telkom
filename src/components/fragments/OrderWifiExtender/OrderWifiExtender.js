import React, { createContext, useEffect, useState } from 'react';
import { Link, useHistory, useParams } from 'react-router-dom';
import Button from '../../../components/elements/Button';
import OtpInput from '../../../components/elements/OtpInput';
import WhiteCardPlain from '../../../components/layouts/WhiteCardPlain';
import WhiteCardSuccessPlain from '../../../components/layouts/WhiteCardSuccessPlain';
import { thousand } from '../../../utils/format';
import styles from './styles.scoped.css';
import useTimer from '../../../hooks/useTimer';

const Context = createContext({});

export default function OrderWifiExtender() {
  const { subpage } = useParams();
  const history = useHistory();

  useEffect(() => {
    subpage && history.replace('/order/wifi-extender');
  }, []);

  return (
    <Context.Provider>
      {(sp => {
        switch (sp) {
          case 'otp': return <Otp />;
          case 'success': return <Success />;
          default: return <Confirm />;
        }
      })(subpage)}
    </Context.Provider>
  );
}

export function Confirm() {
  const history = useHistory();
  const [checked, setChecked] = useState(false);
  const onConfirm = () => {
    history.push('/order/wifi-extender/otp');
  };

  return (
    <WhiteCardPlain back="/shop/internet/wifi-extender" subtitle="Confirm your package selection and order" title="Confirm order">
      <section className={styles.package}>
        <h3>Product</h3>
        <Link to="#">
          <header>
            <figure>
              <img src="/assets/ic_shop_internet_white.png" />
            </figure>
            <h1>Wifi Extender</h1>
            <p>Complete coverage with wifi extender</p>
          </header>
          <footer>
            <h3>Rp{thousand(35000)}<small> /month</small></h3>
            <p>This pricing will replace your internet package fee</p>
          </footer>
        </Link>
        <h3>Total</h3>
        <h1>Rp{thousand(35000)}</h1>
        <p>Amount will be charged to your next bill.<br />Includes 15% tax.</p>
        <label>
          <input checked={checked} onChange={() => setChecked(!checked)} type="checkbox" />
          I agree to the <span>Terms &amp; Conditions</span> and <span>Privacy Policy</span>
        </label>
        <div>
          <Button to="/shop/internet/wifi-extender" variant="bordered">Cancel</Button>
          <Button disabled={!checked} onClick={onConfirm}>Confirm</Button>
        </div>
      </section>
    </WhiteCardPlain>
  );
}

export function Otp() {
  const mobileNumber = '';
  const subtitle = `Please enter 4-digit OTP code sent via SMS to ${mobileNumber.slice(-4).padStart(mobileNumber.length, 'â€¢')}`;
  const initTime = 59;
  const { setTimer, timer, time } = useTimer(initTime);
  const min = time[1];
  const sec = time[2];
  const history = useHistory();

  const onResendCode = () => {
    setTimer(initTime);
  };

  const onSubmit = () => {
    history.push('/order/wifi-extender/success');
  };

  return (
    <WhiteCardPlain back="/order/wifi-extender" icon="verify" subtitle={subtitle} title="Mobile Verification">
      <OtpInput onSubmit={onSubmit} />
      {timer ?
        <p className={styles['verify-timer']}>Resend Code In {min}:{sec}</p> :
        <Button className={styles['verify-resend']} onClick={onResendCode} variant="text">Resend Code</Button>
      }
    </WhiteCardPlain>
  );
}

export function Success() {
  return (
    <WhiteCardSuccessPlain
      icon="success"
      subtitle="Our technician will contact you as soon as possible."
      title="Your Wifi Extender has been successfully requested"
    >
      <Button className={styles['button-success']} to="/">Back to Home</Button>
    </WhiteCardSuccessPlain>
  );
}
