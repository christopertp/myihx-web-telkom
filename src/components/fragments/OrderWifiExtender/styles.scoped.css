.package { margin-bottom: 1.5rem; }

.package > h3 {
  color: #66686a;
  font-size: 0.75rem;
  font-weight: normal;
  letter-spacing: 0.0625rem;
  margin-bottom: 0.5rem;
  text-transform: uppercase;
}

.package > a {
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  display: block;
  overflow: hidden;
  margin-bottom: 1.5rem;
}

.package > a > header {
  background: linear-gradient(0deg, #f2f2f2 0%, #ffffff 100%);
  display: grid;
  padding: 1rem;
  grid-template-columns: auto 1fr;
  grid-template-rows: auto 1fr;
}

.package > a > header > figure {
  align-items: center;
  background: linear-gradient(135deg, #6a120c, #b21f15, #ee4c24, #ff7e00);
  border-radius: 0.5rem;
  color: #ffffff;
  display: flex;
  flex-direction: column;
  grid-row: 1/3;
  height: 5.375rem;
  justify-content: center;
  margin-right: 0.75rem;
  width: 5.375rem;
  padding: 0.5rem;
}
.package > a > header > figure > img {
  width: 100%;
}

.package > a > header > h1 {
  font-size: 1.25rem;
  margin: auto 0 0.25rem;
  transition: color 0.25s;
}

.package > a:hover > header > h1 { color: var(--primary-color); }

.package > a > header > p {
  font-size: 1rem;
  margin-bottom: auto;
}

.package > a > footer { padding: 0.5rem 1rem; }

.package > a > footer > h3 {
  font-size: 1.5rem;
}

.package > a > footer > h3 > small {
  color: #b2b4b5;
  font-size: 1rem;
  font-weight: normal;
}

.package > a > footer > p { color: #66686a; }
.package > h1 { font-size: 1.5rem; }
.package > p { color: #66686a; margin-bottom: 2rem; }

.package > input[type='checkbox'] {
  margin-right: 0.5rem;
  margin-top: 2rem;
}
.package > div {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 1.5rem;
  margin-top: 1rem;
}
.package > label {
  color: #808285;
  display: flex;
  align-items: center;
}
.package > label > span {
  color: #ee3124;
  font-weight: bold;
  padding: 0 0.2rem;
}

.package > label > input {
  appearance: none;
  background-color: #ffffff;
  border: 0.0625rem solid #cccccc;
  border-radius: 0.25rem;
  height: 1.25rem;
  margin-right: 0.5rem;
  position: relative;
  transition: background-color 0.15s;
  width: 1.25rem;
}

.package > label > input:checked {
  background-color: var(--primary-color);
  border: none;
}

.package > label > input:checked::after {
  content: '\2713';
  color: #ffffff;
  line-height: 1.25rem;
  position: absolute;
  text-align: center;
  vertical-align: middle;
  width: 100%;
}

.verify-timer {
  color: #b2b4b5;
  font-size: 0.875rem;
  font-weight: 800;
  margin-top: 2.625rem;
  text-align: center;
  text-transform: uppercase;
}

.verify-resend {
  margin: 2.625rem auto 0;
}

.button-success {
  margin-top: 2rem;
}

@media only screen and (max-width: 48em) {
  .package {
    margin-bottom: 0;
  }
  .package > a > header > figure {
    height: 3.75rem;
    width: 3.75rem;
  }
  .package > a > header > h3 { font-size: 1rem; }
  .package > a > header > p { font-size: 0.75rem; }
  .package > a > footer > h3 { font-size: 1.25rem; }
  .package > a > footer > h3 > small { font-size: 0.75rem; }
  .package > a > footer > p {
    margin-top: 0.25rem;
    font-size: 0.75rem;
  }
  .package > h1 {
    font-size: 1.25rem;
    margin-bottom: 0.25rem;
  }
  .package > p {
    font-size: 0.75rem;
  }
  .package > label {
    font-size: 0.7rem;
  }
  .button-success {
    margin: auto 0 1.5rem;
  }
}
