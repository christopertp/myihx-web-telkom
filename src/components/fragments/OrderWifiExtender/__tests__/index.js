import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import OrderWifiExtender, { Confirm, Otp, Success } from '../OrderWifiExtender';

describe('src/components/fragments/OrderWifiExtender', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<OrderWifiExtender />);
    expect(tree).toMatchSnapshot();
  });

  test('Confirm', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Confirm />);
    expect(tree).toMatchSnapshot();
  });

  test('Otp', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Otp />);
    expect(tree).toMatchSnapshot();
  });

  test('Success', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Success />);
    expect(tree).toMatchSnapshot();
  });
});
