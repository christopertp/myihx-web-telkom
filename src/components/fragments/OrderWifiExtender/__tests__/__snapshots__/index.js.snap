// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/OrderWifiExtender Confirm 1`] = `
<WhiteCardPlain
  back="/shop/internet/wifi-extender"
  close=""
  help={false}
  icon=""
  size="large"
  subtitle="Confirm your package selection and order"
  title="Confirm order"
>
  <section>
    <h3>
      Product
    </h3>
    <Link
      to="#"
    >
      <header>
        <figure>
          <img
            src="/assets/ic_shop_internet_white.png"
          />
        </figure>
        <h1>
          Wifi Extender
        </h1>
        <p>
          Complete coverage with wifi extender
        </p>
      </header>
      <footer>
        <h3>
          Rp
          35.000
          <small>
             /month
          </small>
        </h3>
        <p>
          This pricing will replace your internet package fee
        </p>
      </footer>
    </Link>
    <h3>
      Total
    </h3>
    <h1>
      Rp
      35.000
    </h1>
    <p>
      Amount will be charged to your next bill.
      <br />
      Includes 15% tax.
    </p>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      I agree to the 
      <span>
        Terms & Conditions
      </span>
       and 
      <span>
        Privacy Policy
      </span>
    </label>
    <div>
      <Button
        className=""
        disabled={false}
        onClick={null}
        to="/shop/internet/wifi-extender"
        type="button"
        variant="bordered"
      >
        Cancel
      </Button>
      <Button
        className=""
        disabled={true}
        onClick={[Function]}
        to=""
        type="button"
        variant="red"
      >
        Confirm
      </Button>
    </div>
  </section>
</WhiteCardPlain>
`;

exports[`src/components/fragments/OrderWifiExtender Otp 1`] = `
<WhiteCardPlain
  back="/order/wifi-extender"
  close=""
  help={false}
  icon="verify"
  size="large"
  subtitle="Please enter 4-digit OTP code sent via SMS to "
  title="Mobile Verification"
>
  <OtpInput
    className=""
    disabled={false}
    error=""
    onSubmit={[Function]}
  />
  <p>
    Resend Code In 
    00
    :
    59
  </p>
</WhiteCardPlain>
`;

exports[`src/components/fragments/OrderWifiExtender Success 1`] = `
<WhiteCardSuccessPlain
  close=""
  icon="success"
  subtitle="Our technician will contact you as soon as possible."
  title="Your Wifi Extender has been successfully requested"
>
  <Button
    className=""
    disabled={false}
    onClick={null}
    to="/"
    type="button"
    variant="red"
  >
    Back to Home
  </Button>
</WhiteCardSuccessPlain>
`;

exports[`src/components/fragments/OrderWifiExtender render 1`] = `
<Provider>
  <Confirm />
</Provider>
`;
